CONTRIBUTING.rst

```diff
@@ -0,0 +1,159 @@
+Contributing to the Jupyter Notebook
+====================================
+
+TODO: a welcoming sentence
+
+General Guidelines
+------------------
+
+For general documentation about contributing to Jupyter projects, see the
+`Project Jupyter Contributor Documentation`__.
+
+__ http://jupyter.readthedocs.org/#contributor-documentation
+
+
+Setting Up a Development Environment
+------------------------------------
+
+For general installation instructions have a look at the `Project Jupyter Installation Guide`__.
+
+__ https://jupyter.readthedocs.org/en/latest/install.html
+
+Installing Node.js and npm
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Building the Notebook from its GitHub source code requires some tools to
+create and minify JavaScript components and the CSS.
+
+You can use the pre-built installer from the `Node.js website`__.
+The installer will include Node.js and Node's package manager, ``npm``.
+
+__ https://nodejs.org
+
+Or you can use your system's package manager ...
+
+If you use Homebrew_ on Mac OS X::
+
+    brew install node
+
+.. _Homebrew: http://brew.sh/
+
+For Debian/Ubuntu systems, you should use the ``nodejs-legacy`` package instead
+of the ``node`` package::
+
+    sudo apt-get update
+    sudo apt-get install nodejs-legacy npm
+
+
```

Comment:
Or if you use conda, you can get nodejs and npm with:

`conda install -c javascript nodejs`